src/memory_manager.cpp

```diff
@@ -0,0 +1,204 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+#include "memory_manager.h"
+#include "utils.h"
+
+namespace diskann
+{
+
+size_t MemoryManager::get_memory_used_in_bytes() const
+{
+    return *_memory_used_in_bytes_ptr;
+}
+
+void MemoryManager::alloc_aligned(void **ptr, size_t size, size_t align)
+{
+    *ptr = nullptr;
+    if (IS_ALIGNED(size, align) == 0)
+        report_misalignment_of_requested_size(size, align);
+#ifndef _WINDOWS
+    *ptr = ::aligned_alloc(align, size);
+#else
+    *ptr = ::_aligned_malloc(size, align); // note the swapped arguments!
+#endif
+    if (*ptr == nullptr)
+    {
+        report_memory_allocation_failure(size, align);
+        return;
+    }
+
+    insert_block(*ptr, size);
```

Comment:
can we make this optional? For example, by using it under "#ifdef TRACK_MEM_USAGE"